app/services/foreman/parameters/validator.rb

```diff
@@ -13,6 +13,10 @@ def validate!
           validate_regexp
         when "list", "array"
           validate_list
+        when "hash"
```

Comment:
Sorry, I need to take a step back.  I can't reproduce the original issue described in http://projects.theforeman.org/issues/11862#note-2, and secondly, I can't see how this code is related.

a) Every time I try and reproduce the original issue, I get a validation message:

<pre>
2015-10-22T15:12:02 [app] [I] Started PUT "/puppetclasses/45-activemq" for 127.0.0.1 at 2015-10-22 15:12:02 +0100
2015-10-22T15:12:02 [app] [I] Processing by PuppetclassesController#update as HTML
2015-10-22T15:12:02 [app] [I]   Parameters: {"utf8"=>"✓", "authenticity_token"=>"WA5GDpEyXQDdN06vepyFOBARRgQtB495MWgVbORPMtE=", "puppetclass"=>{"hostgroup_ids"=>["", "39"], "class_params_attributes"=>{....}, "lookup_keys_attributes"=>{"0"=>{"_destroy"=>"false", "key"=>"test2", "description"=>"foobar", "key_type"=>"hash", "default_value"=>"[FI
LTERED]", "validator_type"=>"", "path"=>"fqdn\r\nhostgroup\r\nos\r\ndomain", "merge_overrides"=>"0", "id"=>"220"}}}, "environment_filter"=>"", "commit"=>"»Submit«", "id"=>"45-activemq"}
...
2015-10-22T15:12:24 [app] [I] Failed to save: »Default value« »is invalid«
</pre>

Note that the key_type is hash, I have no validator and no matchers.  The default value should be cast only, no validation.  The error seems to be recorded fine, no exception, and is shown in the UI correctly next to the default value.

b) This particular line is actually in the validator class.  If you trace where it's called from, it's in LookupKey#validate_default_value and the `@options[:type]` is actually LookupKey#validator_type.  That's only ever going to be none, regexp or list.  hash, yaml, json are key_types, not validator_types.